### PR TITLE
[MRG] Refactored path functionality into path.py

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Addresses: #466 
and related to #454 

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [ ] Commit history does not contain any merge commits

### CODE REVIEW

**Comment by adam2392** [comment]
I can rebase after #457 is merged in first(?). 

This was a simple fix and also very necessary in my mind for my brain to find the right functions when I go searching :p. 

Thanks @hoechenberger  for bringing it up.

**Comment by jasmainak** [comment]
Looks pretty okay to me. Can you see why the CIs are failing?

File: mne_bids/tests/test_utils.py [hunk 1]
```diff
@@ -6,14 +6,14 @@
 # License: BSD (3-clause)
```
Comment (agramfort) [inline mne_bids/tests/test_utils.py:6]: functions in path.py should now be tested in test_path.py

File: doc/whats_new.rst [hunk 2]
```diff
@@ -125,9 +126,9 @@ Version 0.3
 Changelog
 ~~~~~~~~~
 
-- New function :func:`mne_bids.utils.get_kinds` for getting data types from a BIDS dataset, by `Stefan Appelhoff`_ (`#253 <https://github.com/mne-tools/mne-bids/pull/253>`_)
-- New function :func:`mne_bids.utils.get_entity_vals` allows to get a list of instances of a certain entity in a BIDS directory, by `Mainak Jas`_ and `Stefan Appelhoff`_ (`#252 <https://github.com/mne-tools/mne-bids/pull/252>`_)
-- :func:`mne_bids.utils.print_dir_tree` now accepts an argument :code:`max_depth` which can limit the depth until which the directory tree is printed, by `Stefan Appelhoff`_ (`#245 <https://github.com/mne-tools/mne-bids/pull/245>`_)
+- New function :func:`mne_bids.path.get_kinds` for getting data types from a BIDS dataset, by `Stefan Appelhoff`_ (`#253 <https://github.com/mne-tools/mne-bids/pull/253>`_)
```
Comment (agramfort) [inline doc/whats_new.rst:129]: are we sure that get_kinds and get_entity_vals are exposed in path and not in the top namespace?

File: doc/whats_new.rst [hunk 3]
```diff
@@ -175,7 +176,7 @@ Changelog
 Bug
 ~~~
 
-- Normalize the length of the branches in :func:`mne_bids.utils.print_dir_tree` by the length of the root path, leading to more adequate visual display, by `Stefan Appelhoff`_ (`#192 <https://github.com/mne-tools/mne-bids/pull/192>`_)
+- Normalize the length of the branches in :func:`mne_bids.path.print_dir_tree` by the length of the root path, leading to more adequate visual display, by `Stefan Appelhoff`_ (`#192 <https://github.com/mne-tools/mne-bids/pull/192>`_)
```
Comment (agramfort) [inline doc/whats_new.rst:179]: idem for print_dir_tree

**Comment by sappelhoff** [comment]
I'll have to pass on reviewing in depth, but I am +1 on the idea.

File: mne_bids/tests/test_path.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
+import os
```
Comment (jasmainak) [inline mne_bids/tests/test_path.py:1]: This file is missing a docstring and author names like the others.

File: mne_bids/tests/test_path.py [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+from os import path as op
```
Comment (jasmainak) [inline mne_bids/tests/test_path.py:2]:
```suggestion
import os.path as op
```

shorter :)

File: doc/api.rst [hunk 6]
```diff
@@ -38,16 +38,16 @@ MNE BIDS
    get_anonymization_daysback
 
 
-Utils
```
Comment (jasmainak) [inline doc/api.rst:41]: everything in `utils` is now in `path`? Or there is no public function in `utils`?

Comment (adam2392) [reply doc/api.rst:41]: Only `get_anonymization_dateback`, which I added back. However, I saw it under the main namespace, so forgot to add it in.

File: doc/api.rst [hunk 7]
```diff
@@ -57,6 +57,7 @@ Utils
    get_kinds
    BIDSPath
```
Comment (agramfort) [inline doc/api.rst:58]: get_kinds is in the top namespace it should nit appear in the path module here

**Comment by jasmainak** [comment]
why are the CIs failing?

File: mne_bids/write.py [hunk 8]
```diff
@@ -33,13 +33,13 @@
...
 from mne_bids.utils import (_write_json, _write_tsv, _write_text,
-                            _read_events, _mkdir_p, _age_on_date,
+                            _read_events, _age_on_date,
                             _infer_eeg_placement_scheme,
-                            _parse_bids_filename, _handle_kind,
-                            _path_to_str, _parse_ext,
-                            _get_ch_type_mapping, make_bids_folders,
-                            _estimate_line_freq, make_bids_basename,
-                            BIDSPath, _check_anonymize, _stamp_to_dt)
+                            _handle_kind, _path_to_str, _get_ch_type_mapping,
```
Comment (jasmainak) [inline mne_bids/write.py:38]: `_path_to_str` should go to path module?

**Comment by adam2392** [comment]
> why are the CIs failing?

Tbh not sure why... I just adjusted some of the imports and now they work!

**Comment by hoechenberger** [comment]
Thanks @adam2392 <3

**Pull request merged**